modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/VerticalNavTag.java

```diff
@@ -162,6 +172,9 @@ else if (_active == null) {
 			if (expanded == null) {
 				expanded = Boolean.FALSE;
 			}
+			else {
+				setDefaultExpandedKeys((String)verticalNavItem.get("id"));
+			}
```

Comment:
good trick, doing it at render time shall save some cpu cycles when preparing the props!

Reply:
Please see my comment on `itemActive`, I don't think it's right to set the attribute value in the rendering logic.